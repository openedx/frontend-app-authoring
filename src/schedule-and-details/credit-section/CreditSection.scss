.credit-section .credit-info-list {
  display: flex;
  justify-content: space-between;
  margin: 1.25rem 0 1.75rem;
  padding-inline-start: 0;
  list-style-type: none;
}
