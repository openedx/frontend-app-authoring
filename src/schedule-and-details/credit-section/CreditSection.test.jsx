import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
// eslint-disable-next-line import/no-extraneous-dependencies
import { IntlProvider } from 'react-intl';

import CreditSection from '.';

describe('<CreditSection />', () => {
  const RootWrapper = (props) => (
    <IntlProvider locale="en">
      <CreditSection {...props} />
    </IntlProvider>
  );

  const props = {
    creditRequirements: {
      grade: [
        {
          name: 'grade',
          displayName: 'Minimum Grade',
          criteria: {
            minGrade: 0.8,
          },
        },
      ],
    },
  };

  it('matches the snapshots', () => {
    const tree = renderer.create(<RootWrapper {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders credit section successfully', () => {
    const { getByText } = render(<RootWrapper {...props} />);
    expect(getByText(/Course Credit Requirements/i)).toBeInTheDocument();
    expect(
      getByText(/Steps required to earn course credit/i),
    ).toBeInTheDocument();
    expect(
      getByText(
        /A requirement appears in this list when you publish the unit that contains the requirement./i,
      ),
    ).toBeInTheDocument();
  });

  it('renders empty requirements', () => {
    const initialProps = { ...props, creditRequirements: {} };
    const { getByText } = render(<RootWrapper {...initialProps} />);
    expect(getByText(/No credit requirements found./i)).toBeInTheDocument();
  });

  it('renders requirements successfully', () => {
    const { getByText, queryAllByText } = render(<RootWrapper {...props} />);
    expect(getByText(/Minimum Grade/)).toBeInTheDocument();
    expect(getByText(/80%/i)).toBeInTheDocument();
    expect(queryAllByText('Successful Proctored Exam').length).toBe(0);
    expect(queryAllByText('ID Verification').length).toBe(0);
  });
});
