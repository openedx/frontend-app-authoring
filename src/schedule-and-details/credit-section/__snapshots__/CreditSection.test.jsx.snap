// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreditSection /> matches the snapshots 1`] = `
<section
  className="section-container credit-section"
>
  <header
    className="section-header"
  >
    <span
      className="lead"
    >
      Course Credit Requirements
    </span>
    <span
      className="x-small text-gray-700"
    >
      Steps required to earn course credit
    </span>
  </header>
  <span>
    A requirement appears in this list when you publish the unit that contains the requirement.
  </span>
  <ul
    className="credit-info-list"
  >
    <li
      className="d-grid"
    >
      <h4
        className="text-gray-700"
      >
        Minimum Grade
      </h4>
      <div
        className="d-flex flex-column"
      >
        <span
          className="small"
        >
          80%
        </span>
      </div>
    </li>
  </ul>
</section>
`;
